update_checkout: add mimalloc

compnerd opened the pull request:
Add mimalloc to the checkout set. This is to allow us to build the project for the Windows toolchain.

compnerd committed:
update_checkout: add mimalloc

Add mimalloc to the checkout set. This is to allow us to build the
project for the Windows toolchain.

compnerd commented:
@swift-ci please smoke test

shahmishal commented on utils/update_checkout/update-checkout-config.json:
```diff
@@ -167,7 +170,8 @@
                 "libxml2": "v2.11.5",
-                "zlib": "v1.3.1"
+                "zlib": "v1.3.1",
+                "mimalloc": "v3.0.1"
```
Can you add this to the next and rebranch scheme too?

compnerd replied:
Whoops, absolutely!

compnerd commented:
@swift-ci please smoke test

compnerd commented:
@swift-ci please smoke test macOS platform

compnerd commented:
@swift-ci please smoke test Linux platform

compnerd commented:
@swift-ci please smoke test macOS platform

etcwilde commented on utils/update_checkout/update-checkout-config.json:
```diff
@@ -113,6 +113,9 @@
           "remote": { "id": "madler/zlib" }
+        },
+        "mimalloc": {
+          "remote": { "id": "microsoft/mimalloc" }
```
Can you add
```suggestion
          "remote": { "id": "microsoft/mimalloc" },
          "platforms": [ "Windows" ]
```

This way we're only downloading it on Windows.

compnerd replied:
I think that is best done in a follow up if we want to remove some of these repositories from being checked out (e.g. playground-support).

etcwilde replied:
I'm fine if there's a follow-up for the others, but this is new stuff.

compnerd replied:
Doing that as a single pass is better IMO.

compnerd committed:
utils: remove some repositories from the default checkout

Only code xcode-playground-support on Darwin, mimalloc on Windows.

compnerd committed:
utils: remove `icu` checkout on rebranch

The ICU repository is no longer used, supplanted by
swift-foundation-icu. Remove the unnecessary clone to save some time.

compnerd commented:
@swift-ci please test

etcwilde commented:
@swift-ci please test macOS

etcwilde commented:
@swift-ci please test macOS
